src/nccl_ofi_rdma.c

```diff
@@ -249,6 +249,16 @@ static inline nccl_net_ofi_ep_rail_t *get_rail(nccl_net_ofi_rdma_ep_t *ep,
 	return &ep->rails[rail_id];
 }
 
+/*
+ * @brief return the domain for the endpoint and rail.
+ */
+
+static inline struct fid_domain *get_domain_from_endpoint(nccl_net_ofi_ep_t *base_ep, int rail_id)
```

Comment:
Nit: Wouldn't it make sense for `get_domain_from_endpoint` to take a `nccl_net_ofi_rdma_ep_t` as an argument and avoid the cast below? Similarly for SENDRECV protocol.

Reply:
Yeah, fixed.  I was initially trying to make this generic.